src/databricks/labs/ucx/hive_metastore/pipelines_migrate.py

```diff
@@ -0,0 +1,142 @@
+import logging
+from functools import partial
+from typing import BinaryIO
+
+from databricks.labs.blueprint.parallel import Threads
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import DatabricksError
+from databricks.sdk.service.jobs import PipelineTask, Task, JobSettings
+
+from databricks.labs.ucx.assessment.pipelines import PipelineInfo, PipelinesCrawler
+
+logger = logging.getLogger(__name__)
+
+
+class PipelinesMigrator:
+    def __init__(
+        self, ws: WorkspaceClient, pipelines_crawler: PipelinesCrawler, catalog_name: str, skip_pipelines=None
+    ):
+        self._ws = ws
+        self._pipeline_crawler = pipelines_crawler
+        self._catalog_name = catalog_name
+        self._skip_pipelines = skip_pipelines or []
+        self._pipeline_job_tasks_mapping: dict[str, list[dict]] = {}
+
+    def _populate_pipeline_job_tasks_mapping(self) -> None:
+        jobs = self._ws.jobs.list()
+
+        for job in jobs:
+            if not job.job_id:
+                continue
+
+            job_details = self._ws.jobs.get(job.job_id)
+            if not job_details.settings or not job_details.settings.tasks:
+                continue
+
+            for task in job_details.settings.tasks:
+                if not task.pipeline_task:
+                    continue
+                pipeline_id = task.pipeline_task.pipeline_id
+                job_info = {"job_id": job.job_id, "task_key": task.task_key}
+                if pipeline_id not in self._pipeline_job_tasks_mapping:
+                    self._pipeline_job_tasks_mapping[pipeline_id] = [job_info]
+                else:
+                    self._pipeline_job_tasks_mapping[pipeline_id].append(job_info)
+                logger.info(f"Found job:{job.job_id} task:{task.task_key} associated with pipeline {pipeline_id}")
+
+    def _get_pipelines_to_migrate(self) -> list[PipelineInfo]:
+        return list(self._pipeline_crawler.snapshot())
+
+    def migrate_pipelines(self) -> None:
```

Comment:
Add docs or mention this in the class docs

Reply:
Maybe `PipelinesMigrator.migrate` makes more sense, depends on how other migrator classes do it